recipe/build.sh

```diff
@@ -1,39 +1,88 @@
+GLX_OPTION="-Dglx=disabled"
+EGL_OPTION="-Degl=disabled"
+GBM_OPTION="-Dgbm=disabled"
+GALLIUM_DRIVERS="-Dgallium-drivers=softpipe,llvmpipe"
+VULKAN_DRIVERS="-Dvulkan-drivers=swrast"
+
+# glvnd must be disabled if both GLX and EGL are disabled; Meson requires at least one for glvnd support.
+# This matches conda-forge and modern best practices: let the system or other packages provide glvnd.
+GLVND_OPTION="-Dglvnd=disabled"
+
+# Set platforms option based on target platform
+if [[ "$target_platform" == osx* ]]; then
+  PLATFORMS="-Dplatforms=macos"
+else
+  PLATFORMS="-Dplatforms=x11"
 fi
 
-export PKG_CONFIG_PATH="${PREFIX}/lib/pkgconfig/:"${PKG_CONFIG_PATH}
-export CFLAGS="-I${PREFIX}/include "${CFLAGS}
-export LDFLAGS="-L${PREFIX}/lib "${LDFLAGS}
-
-./configure \
-    --prefix=$PREFIX \
-    --disable-gles1 \
-    --disable-gles2 \
-    --disable-va \
-    --disable-gbm \
-    --disable-xvmc \
-    --disable-vdpau \
-    --enable-shared-glapi \
-    --enable-texture-float \
-    --disable-dri \
-    --with-dri-drivers= \
-    --with-gallium-drivers=swrast \
-    --disable-egl \
-    --with-egl-platforms= \
-    --enable-gallium-osmesa \
-    --disable-glx \
-    --enable-llvm \
-    --disable-llvm-shared-libs \
-    --with-osmesa-bits=32
-
-make -j${CPU_COUNT}
-make install
+COMMON_OPTIONS="-Dzstd=enabled -Dopengl=true -Dtools=[] -Dbuild-tests=true"
+LINUX_OPTIONS="-Dlibunwind=enabled -Dshared-glapi=enabled"
+OSMESA_OPTIONS="-Dosmesa=true"
+OPENCL_OPTIONS="-Dgallium-opencl=disabled"
+
+# OSMesa options - Mesa 25.1.0 still supports OSMesa but the interface has changed
+OSMESA_OPTIONS="-Dosmesa=true"
+
+# ---
+# Meson options/choices and rationale:
+# -Dgallium-drivers: Comma-separated list of Gallium drivers to build (e.g. softpipe,llvmpipe,zink)
+# -Dglx: Valid values are 'auto', 'disabled', 'dri', 'xlib'. Use 'dri' for Linux, 'disabled' for macOS.
+# -Degl: Enable or disable EGL support. Disabled on macOS, enabled on Linux.
+# -Dplatforms: Comma-separated list of platforms (e.g. x11,macos,wayland,drm). Use 'x11' for Linux, 'macos' for macOS.
+# -Dvulkan-drivers: Comma-separated list of Vulkan drivers (e.g. swrast,virtio,all). Use 'swrast,virtio' for Linux, 'swrast' for macOS.
+# -Dlibunwind: Enable or disable libunwind support. Enabled on Linux, disabled on macOS.
+# -Dshared-glapi: Enable shared GL API. Enabled for both Linux and macOS.
+#
+# These options were chosen to match upstream, conda-forge, and downstream needs, and to avoid building unsupported or unnecessary features on each platform.
+# ---
+
+# The --prefix=$PREFIX flag ensures proper installation location for conda-build
+meson setup builddir/ \
+  ${MESON_ARGS} \
+  --prefix=$PREFIX \
+  -Dlibdir=lib \
+  $PLATFORMS \
+  $VULKAN_DRIVERS \
+  $GALLIUM_DRIVERS \
+  -Dgallium-va=disabled \
+  -Dgallium-vdpau=disabled \
+  -Dgles1=disabled \
+  -Dgles2=disabled \
+  $GBM_OPTION \
+  $GLVND_OPTION \
+  $GLX_OPTION \
+  $EGL_OPTION \
+  -Dllvm=enabled \
+  -Dshared-llvm=enabled \
```

Comment:
```suggestion
  -Dshared-llvm=enabled \
```
We might want to have a static link here instead. Similar to https://github.com/AnacondaRecipes/llvmlite-feedstock/pull/15

Reply:
I set `-Dshared-llvm=false`

Reply:
I'm getting the error:

[1036/1045] Linking target src/gallium/drivers/llvmpipe/lp_test_format
```
FAILED: src/gallium/drivers/llvmpipe/lp_test_format

$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-c++  -o src/gallium/drivers/llvmpipe/lp_test_format src/gallium/drivers/llvmpipe/lp_test_format.p/lp_test_format.c.o src/gallium/drivers/llvmpipe/lp_test_format.p/lp_test_main.c.o -L$PREFIX/lib -Wl,--as-needed -Wl,--no-undefined -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,-rpath,$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/mesalib-25.1.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem $PREFIX/include -Wl,--start-group src/gallium/drivers/llvmpipe/libllvmpipe.a src/util/libmesa_util.a src/util/libmesa_util_sse41.a src/util/blake3/libblake3.a src/c11/impl/libmesa_util_c11.a src/gallium/auxiliary/libgallium.a src/compiler/nir/libnir.a src/compiler/libcompiler.a -lrt -ldl -lpthread -lm -L$PREFIX/lib -lLLVMLTO -lLLVMPasses -lLLVMObjCARCOpts -lLLVMExtensions -lLLVMCoroutines -lLLVMipo -lLLVMVectorize -lLLVMLinker -lLLVMIRReader -lLLVMAsmParser -lLLVMFrontendOpenMP -lLLVMX86TargetMCA -lLLVMMCA -lLLVMX86Disassembler -lLLVMX86AsmParser -lLLVMX86CodeGen -lLLVMCFGuard -lLLVMGlobalISel -lLLVMX86Desc -lLLVMX86Info -lLLVMSelectionDAG -lLLVMInstrumentation -lLLVMAsmPrinter -lLLVMMCJIT -lLLVMMCDisassembler -lLLVMInterpreter -lLLVMExecutionEngine -lLLVMRuntimeDyld -lLLVMOrcTargetProcess -lLLVMOrcShared -lLLVMCodeGen -lLLVMTarget -lLLVMScalarOpts -lLLVMInstCombine -lLLVMAggressiveInstCombine -lLLVMTransformUtils -lLLVMBitWriter -lLLVMAnalysis -lLLVMProfileData -lLLVMSymbolize -lLLVMDebugInfoPDB -lLLVMDebugInfoMSF -lLLVMDebugInfoDWARF -lLLVMObject -lLLVMTextAPI -lLLVMMCParser -lLLVMMC -lLLVMDebugInfoCodeView -lLLVMBitReader -lLLVMCore -lLLVMRemarks -lLLVMBitstreamReader -lLLVMBinaryFormat -lLLVMSupport -lLLVMDemangle -lz -lzstd -lxml2 -pthread $PREFIX/lib/libz.so $PREFIX/lib/libzstd.so $PREFIX/lib/libunwind.so $PREFIX/lib/libdrm.so -L$PREFIX/lib -lLLVMLTO -lLLVMPasses -lLLVMObjCARCOpts -lLLVMExtensions -lLLVMCoroutines -lLLVMipo -lLLVMVectorize -lLLVMLinker -lLLVMIRReader -lLLVMAsmParser -lLLVMFrontendOpenMP -lLLVMX86TargetMCA -lLLVMMCA -lLLVMX86Disassembler -lLLVMX86AsmParser -lLLVMX86CodeGen -lLLVMCFGuard -lLLVMGlobalISel -lLLVMX86Desc -lLLVMX86Info -lLLVMSelectionDAG -lLLVMInstrumentation -lLLVMAsmPrinter -lLLVMMCJIT -lLLVMMCDisassembler -lLLVMInterpreter -lLLVMExecutionEngine -lLLVMRuntimeDyld -lLLVMOrcTargetProcess -lLLVMOrcShared -lLLVMCodeGen -lLLVMTarget -lLLVMScalarOpts -lLLVMInstCombine -lLLVMAggressiveInstCombine -lLLVMTransformUtils -lLLVMBitWriter -lLLVMAnalysis -lLLVMProfileData -lLLVMSymbolize -lLLVMDebugInfoPDB -lLLVMDebugInfoMSF -lLLVMDebugInfoDWARF -lLLVMObject -lLLVMTextAPI -lLLVMMCParser -lLLVMMC -lLLVMDebugInfoCodeView -lLLVMBitReader -lLLVMCore -lLLVMRemarks -lLLVMBitstreamReader -lLLVMBinaryFormat -lLLVMSupport -lLLVMDemangle -lz -lzstd -lxml2 -L$PREFIX/lib -lLLVMLTO -lLLVMPasses -lLLVMObjCARCOpts -lLLVMExtensions -lLLVMCoroutines -lLLVMipo -lLLVMVectorize -lLLVMLinker -lLLVMIRReader -lLLVMAsmParser -lLLVMFrontendOpenMP -lLLVMX86TargetMCA -lLLVMMCA -lLLVMX86Disassembler -lLLVMX86AsmParser -lLLVMX86CodeGen -lLLVMCFGuard -lLLVMGlobalISel -lLLVMX86Desc -lLLVMX86Info -lLLVMSelectionDAG -lLLVMInstrumentation -lLLVMAsmPrinter -lLLVMMCJIT -lLLVMMCDisassembler -lLLVMInterpreter -lLLVMExecutionEngine -lLLVMRuntimeDyld -lLLVMOrcTargetProcess -lLLVMOrcShared -lLLVMCodeGen -lLLVMTarget -lLLVMScalarOpts -lLLVMInstCombine -lLLVMAggressiveInstCombine -lLLVMTransformUtils -lLLVMBitWriter -lLLVMAnalysis -lLLVMProfileData -lLLVMSymbolize -lLLVMDebugInfoPDB -lLLVMDebugInfoMSF -lLLVMDebugInfoDWARF -lLLVMObject -lLLVMTextAPI -lLLVMMCParser -lLLVMMC -lLLVMDebugInfoCodeView -lLLVMBitReader -lLLVMCore -lLLVMRemarks -lLLVMBitstreamReader -lLLVMBinaryFormat -lLLVMSupport -lLLVMDemangle -lz -lzstd -lxml2 -Wl,--end-group

$BUILD_PREFIX/bin/../lib/gcc/x86_64-conda-linux-gnu/11.2.0/../../../../x86_64-conda-linux-gnu/bin/ld: src/gallium/auxiliary/libgallium.a.p/gallivm_lp_bld_misc.cpp.o: in function `lp_build_create_jit_compiler_for_module':

/usr/local/src/conda/mesalib-25.1.0/builddir/../src/gallium/auxiliary/gallivm/lp_bld_misc.cpp:612: undefined reference to `llvm::JITEventListener::createIntelJITEventListener()'
```

Reply:
conda-forge builds shared libraries, see https://github.com/regro-cf-autotick-bot/mesalib-feedstock/blob/25.1.0_h693059/recipe/build.sh#L38-L39.

The flag `-Dshared-llvm=true` with `libllvm15` in `run` (without it will be overlinking error for libvulkan_lvp.dylib and gallium-pipe/pipe_swrast.dylib) solves the issue.